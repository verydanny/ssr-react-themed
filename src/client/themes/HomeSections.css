@value ( RentBlue, RentBlueDark, UniversalGrey ) from 'ui/shared/themes/colors/colors.css';

.HomeSections h2 {
  font-size: 32px;
  font-weight: 900;
  color: RentBlueDark;
  margin: 0 21px;
  padding: 39px 0 12px 0;
}

.HomeSections_SectionBody {
  font-size: 16px;
  line-height: 22px;
  color: UniversalGrey;
  margin: 0 21px;
}

.HomeSections_SectionDynamicLinks,
.HomeSections_SectionLinks ul {
  list-style: none;
  padding: 0;
  margin: 10px 21px 0 21px;
}

.HomeSections_SectionDynamicLinks li,
.HomeSections_SectionLinks li {
  margin-top: 24px;
}

.HomeSections_SectionDynamicLinks a,
.HomeSections_SectionLinks a {
  font-size: 16px;
  font-weight: 700;
  color: RentBlue;
  text-decoration: none;
}
