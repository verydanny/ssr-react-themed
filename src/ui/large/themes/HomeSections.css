@value ( RentBlue, RentBlueDark, UniversalGrey ) from 'ui/shared/themes/colors/colors.css';

.HomeSections {
  max-width: 1023px;
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  flex-flow: row wrap;
  justify-content: space-between;
}

.HomeSections h2 {
  font-size: 32px;
  font-weight: 900;
  color: RentBlueDark;
  margin: 0 21px;
  padding: 64px 0 12px 0;
}

.HomeSections_Section {
  width: 470px;
}

.HomeSections_SectionBody {
  font-size: 16px;
  line-height: 22px;
  color: UniversalGrey;
  margin: 0 21px;
}

.HomeSections_SectionDynamicLinks,
.HomeSections_SectionLinks ul {
  list-style: none;
  padding: 0;
  margin: 10px 21px 0 21px;
}

.HomeSections_SectionDynamicLinks li,
.HomeSections_SectionLinks li {
  margin-top: 24px;
}

.HomeSections_SectionDynamicLinks a,
.HomeSections_SectionLinks a {
  font-size: 16px;
  font-weight: 700;
  color: RentBlue;
  text-decoration: none;
}
