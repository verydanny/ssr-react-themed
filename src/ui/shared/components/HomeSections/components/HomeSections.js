import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import themed from 'react-themed-too'
import autobind from 'autobind-decorator'

@themed(/^HomeSections/, { pure: true })
export default class HomeSections extends PureComponent {
  static propTypes = {
    sections: PropTypes.arrayOf(PropTypes.shape({
      headline: PropTypes.string.isRequired,
      body: PropTypes.string,
      links: PropTypes.string.isRequired,
    })),
    searchTerm: PropTypes.string,
    petFriendlyLink: PropTypes.string,
    studioLink: PropTypes.string,
    oneBedLink: PropTypes.string,
    twoBedLink: PropTypes.string,
    threeBedLink: PropTypes.string,
    cheapLink: PropTypes.string,
    theme: PropTypes.object,
  }

  static defaultProps = {
    theme: {},
  }

  @autobind
  firstTimeLinks() {
    const { searchTerm, studioLink, oneBedLink } = this.props

    if (!searchTerm || !studioLink || !oneBedLink) {
      return null
    }

    return [
      <a href={studioLink}>Find Studio Apartments in {searchTerm}</a>,
      <a href={oneBedLink}>Find 1 Bedroom Apartments in {searchTerm}</a>,
    ]
  }

  @autobind
  petOwnerLinks() {
    const { searchTerm, petFriendlyLink } = this.props

    if (searchTerm && petFriendlyLink) {
      return [
        <a href={petFriendlyLink}>Find Pet Friendly Apartments in {searchTerm}</a>,
      ]
    }

    return [
      <a href="/pet-friendly-apartments">Find Pet Friendly Apartments</a>,
    ]
  }

  @autobind
  studentLinks() {
    const { searchTerm, twoBedLink, threeBedLink } = this.props

    if (searchTerm && twoBedLink && threeBedLink) {
      return [
        <a href={twoBedLink}>Find 2 Bedroom Apartments in {searchTerm}</a>,
        <a href={threeBedLink}>Find 3 Bedroom Apartments in {searchTerm}</a>,
      ]
    }

    return [
      <a href="/off-campus-housing">Off Campus Housing</a>,
    ]
  }

  @autobind
  budgetLinks() {
    const { searchTerm, cheapLink } = this.props

    if (searchTerm && cheapLink) {
      return [
        <a href={cheapLink}>Find Cheap Apartments in {searchTerm}</a>,
      ]
    }

    return [
      <a href="/cheap-apartments-for-rent">Find Cheap Apartments</a>,
    ]
  }

  dynamicLinks(section) {
    const { theme } = this.props
    // These indices are hardcoded to correspond to specific sections of the SEO Fusion content.
    const linksFn = {
      0: this.firstTimeLinks,
      2: this.petOwnerLinks,
      3: this.studentLinks,
      4: this.budgetLinks,
    }
    const links = linksFn[section] ? linksFn[section]() : null

    if (!links) return null

    return (
      <ul className={theme.HomeSections_SectionDynamicLinks}>
        {links.map((link, index) => <li key={`link_${index}`}>{link}</li>)}
      </ul>
    )
  }

  renderSections() {
    const { sections, theme } = this.props

    /* eslint-disable react/no-danger */
    return sections.map(({ headline, body, links }, index) => (
      <div
        className={theme.HomeSections_Section}
        key={`HomeSection_${index}`}
        data-tid={`home-section-${index}`}
      >
        <h2>{headline}</h2>
        <p className={theme.HomeSections_SectionBody}>{body}</p>
        {this.dynamicLinks(index)}
        <div
          className={theme.HomeSections_SectionLinks}
          dangerouslySetInnerHTML={{ __html: links }}
        />
      </div>
    ))
  }

  render() {
    const { sections, theme } = this.props

    if (!sections || !sections.length) return null

    return (
      <div
        className={theme.HomeSections}
        data-tid="home-sections-container"
      >
        {this.renderSections()}
      </div>
    )
  }
}
