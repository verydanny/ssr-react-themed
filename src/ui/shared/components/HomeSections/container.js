import { connect } from 'react-redux'
import { getHomeSections } from 'app/selectors/meta'
import {
  getSearchTerm,
  getPetFriendlyLink,
  getStudioApartmentsLink,
  getOneBedApartmentsLink,
  getTwoBedApartmentsLink,
  getThreeBedApartmentsLink,
  getCheapApartmentsLink,
} from 'app/selectors/defaultSearch'
import HomeSections from './components/HomeSections'

const mapStateToProps = state => ({
  sections: getHomeSections(state),
  searchTerm: getSearchTerm(state),
  petFriendlyLink: getPetFriendlyLink(state),
  studioLink: getStudioApartmentsLink(state),
  oneBedLink: getOneBedApartmentsLink(state),
  twoBedLink: getTwoBedApartmentsLink(state),
  threeBedLink: getThreeBedApartmentsLink(state),
  cheapLink: getCheapApartmentsLink(state),
})

export default connect(mapStateToProps, {})(HomeSections)
