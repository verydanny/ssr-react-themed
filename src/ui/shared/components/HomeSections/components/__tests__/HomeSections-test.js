import React from 'react'
import { shallow } from 'enzyme'
import HomeSections from '../HomeSections'

const testSections = [
  {
    headline: 'Test Headline 1',
    body: 'Test Body 1',
    links: 'Test Links 1',
  },
  {
    headline: 'Test Headline 2',
    body: 'Test Body 2',
    links: 'Test Links 2',
  },
  {
    headline: 'Test Headline 3',
    body: 'Test Body 3',
    links: 'Test Links 3',
  },
  {
    headline: 'Test Headline 4',
    body: 'Test Body 4',
    links: 'Test Links 4',
  },
  {
    headline: 'Test Headline 5',
    body: 'Test Body 5',
    links: 'Test Links 5',
  },
  {
    headline: 'Test Headline 6',
    body: 'Test Body 6',
    links: 'Test Links 6',
  },
]

const theme = {
  HomeSections_Section: 'section',
  HomeSections_SectionLinks: 'links',
}

const component = () => (
  <HomeSections
    sections={testSections}
    searchTerm="Atlanta, GA"
    petFriendlyLink="/georgia/atlanta-apartments/pet-friendly"
    studioLink="/georgia/atlanta-apartments/studio"
    oneBedLink="/georgia/atlanta-apartments/1-bedroom"
    twoBedLink="/georgia/atlanta-apartments/2-bedroom"
    threeBedLink="/georgia/atlanta-apartments/3-bedroom"
    cheapLink="/georgia/atlanta-apartments/cheap"
    theme={theme}
  />
)

describe('ui/shared/components/HomeSections', () => {
  it('matches the snapshot of sections rendered', () => {
    const wrapper = shallow(component())
    expect(wrapper.html()).toMatchSnapshot()
  })
})
