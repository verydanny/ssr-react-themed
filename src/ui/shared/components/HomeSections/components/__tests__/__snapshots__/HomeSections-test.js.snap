// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/shared/components/HomeSections matches the snapshot of sections rendered 1`] = `

<div data-tid="home-sections-container">
  <div class="section"
       data-tid="home-section-0"
  >
    <h2>
      Test Headline 1
    </h2>
    <p>
      Test Body 1
    </p>
    <ul>
      <li>
        <a href="/georgia/atlanta-apartments/studio">
          Find Studio Apartments in Atlanta, GA
        </a>
      </li>
      <li>
        <a href="/georgia/atlanta-apartments/1-bedroom">
          Find 1 Bedroom Apartments in Atlanta, GA
        </a>
      </li>
    </ul>
    <div class="links">
      Test Links 1
    </div>
  </div>
  <div class="section"
       data-tid="home-section-1"
  >
    <h2>
      Test Headline 2
    </h2>
    <p>
      Test Body 2
    </p>
    <div class="links">
      Test Links 2
    </div>
  </div>
  <div class="section"
       data-tid="home-section-2"
  >
    <h2>
      Test Headline 3
    </h2>
    <p>
      Test Body 3
    </p>
    <ul>
      <li>
        <a href="/georgia/atlanta-apartments/pet-friendly">
          Find Pet Friendly Apartments in Atlanta, GA
        </a>
      </li>
    </ul>
    <div class="links">
      Test Links 3
    </div>
  </div>
  <div class="section"
       data-tid="home-section-3"
  >
    <h2>
      Test Headline 4
    </h2>
    <p>
      Test Body 4
    </p>
    <ul>
      <li>
        <a href="/georgia/atlanta-apartments/2-bedroom">
          Find 2 Bedroom Apartments in Atlanta, GA
        </a>
      </li>
      <li>
        <a href="/georgia/atlanta-apartments/3-bedroom">
          Find 3 Bedroom Apartments in Atlanta, GA
        </a>
      </li>
    </ul>
    <div class="links">
      Test Links 4
    </div>
  </div>
  <div class="section"
       data-tid="home-section-4"
  >
    <h2>
      Test Headline 5
    </h2>
    <p>
      Test Body 5
    </p>
    <ul>
      <li>
        <a href="/georgia/atlanta-apartments/cheap">
          Find Cheap Apartments in Atlanta, GA
        </a>
      </li>
    </ul>
    <div class="links">
      Test Links 5
    </div>
  </div>
  <div class="section"
       data-tid="home-section-5"
  >
    <h2>
      Test Headline 6
    </h2>
    <p>
      Test Body 6
    </p>
    <div class="links">
      Test Links 6
    </div>
  </div>
</div>

`;
